### ISSUE TITLE
Problem in all code execution for CH.8

### ISSUE THREAD

**Issue by Psiopteryx** [PacktPublishing/Deep-Reinforcement-Learning-Hands-On-Second-Edition#30]
Hello,
Loving the book, but in Ch. 8 receive this error for all scripts:
Traceback (most recent call last):
  File "C:/Users/hj/PycharmProjects/DeepLearning/01_dqn_basic.py", line 62, in <module>
    common.setup_ignite(engine, params, exp_source, NAME)
  File "C:\Users\hj\PycharmProjects\DeepLearning\lib\common.py", line 158, in setup_ignite
    handler.attach(engine)
  File "C:\Users\hj\Anaconda3\envs\torch13\lib\site-packages\ptan\ignite.py", line 35, in attach
    engine.register_events(*EpisodeEvents)
  File "C:\Users\hj\Anaconda3\envs\torch13\lib\site-packages\ignite\engine\engine.py", line 223, in register_events
    "Value at {} of event_names should be a str or EventEnum, but given {}".format(index, e)
TypeError: Value at 0 of event_names should be a str or EventEnum, but given EpisodeEvents.EPISODE_COMPLETED

I'm running torch 1.3, and Ignite 0.4.2, and PTAN 0.6

**Comment by vfdev-5** [comment]
Hi, this is due to newer version of ignite 0.4.2.
Please, use the version from requirements or 0.3.0. They should be OK.

**Comment by Psiopteryx** [comment]
Thanks for getting back so quickly; it now executes but there's another
small bug,
NotADirectoryError: [WinError 267] The directory name is invalid:
'runs/2020-10-13T18:28-pong-01_baseline'

I'll work on that one unless you've come across it before!

**Comment by Shmuma** [comment]
I guess changing it to the 
```
now = datetime.now().isoformat(timespec='minutes').replace(':', '')
```
should be enough

**Comment by Psiopteryx** [comment]
Thank you, worked a treat.

Ignite's now complaining (perhaps this more Windows quirkiness):
Traceback (most recent call last):
  File "C:/Users/hj/PycharmProjects/DeepLearning/03_dqn_double.py", line
101, in <module>
    engine.run(common.batch_generator(buffer, params.replay_initial,
params.batch_size))
  File
"C:\Users\hj\Anaconda3\envs\torch13\lib\site-packages\ignite\engine\engine.py",
line 835, in run
    raise ValueError("Argument `epoch_length` should be defined if `data`
is an iterator")
ValueError: Argument `epoch_length` should be defined if `data` is an
iterator

**Comment by vfdev-5** [comment]
@Psiopteryx well, you have to use the version from the requirements: pytorch-ignite==0.2.1

**Comment by vfdev-5** [comment]
@Shmuma any plans to update the code to the latest ignite version ?

**Comment by Psiopteryx** [comment]
Thank you again, works fine under 0.2.1. Had thought 0.3 was fine for
Ignite.

**Comment by Shmuma** [comment]
@vfdev-5 yep, want to update all the samples to recent versions, but have been procastinating this for a while :)

**Comment by AndreasGronlund** [comment]
In ptan:

A) In the file https://github.com/Shmuma/ptan/blob/master/ptan/ignite.py
1. Make the import
from ignite.engine import EventEnum

2. Change
line 10: class EpisodeEvents(enum.Enum): -> class EpisodeEvents(EventEnum):
line 106: class PeriodEvents(enum.Enum): -> class PeriodEvents(EventEnum):

B) Change requirements to the latest versions of torch and torch-ignite In the files https://github.com/Shmuma/ptan/blob/master/requirements.txt and https://github.com/Shmuma/ptan/blob/master/setup.py

**Comment by Shmuma** [comment]
@AndreasGronlund Thanks! Finally started working on pytorch 1.7 support, could be tracked (and contributed) here: https://github.com/Shmuma/ptan/tree/torch-1.7 and here: https://github.com/PacktPublishing/Deep-Reinforcement-Learning-Hands-On-Second-Edition/tree/torch-1.7 :)

**Comment by Shmuma** [comment]
This issue was fixed in branches `torch-1.7` in book repo and ptan. New ptan package haven't been released yet, so, you'll need to setup it from branch, but examples up to chapter 8 was tested on torch-1.7 and ignite-0.4.2
